modules/ROOT/pages/dataweave-extension-plugin.adoc

```diff
@@ -0,0 +1,341 @@
+│       │       └── default
+│       │           └── inputs
+│       │               └── payload.json //[Sample data](#sample-data) for your mappings
+│       ├── resources
+├── pom.xml
+```
+
+=== Create a DataWeave Project
+
+Follow these steps to create a new project:
+
+. Open Visual Studio Code.
+. Launch *Command Palette* by pressing `Ctrl + Shift + P` (Windows, Linux) or `⇧⌘P` (macOS).
+. In the search bar, type `dataweave`.
+. Select *DataWeave: Create New Library Project*.
++
+image::dataweave-extension-new-project.png[Command Palette with DataWeave: Create New Library Project option selected]
+[start=5]
+. Follow prompts to complete an initial configuration for your project:
++
+.. *Organization ID* (Group ID)
+.. *Artifact ID*
+.. *Version*
+.. *Project name*
+[start=5]
+. Choose a directory folder for your new project.
+. Select *OK*.
+. When prompted, select *Yes* or *No* whether you want to open the project in a new Visual Studio Code window.
+
+The following image shows a new DataWeave project in Visual Studio Code:
+
+image::dataweave-extension-new-project-sample.png[DataWeave project example window]
+
+//image::animations/new_project.gif[New project creation steps]
+//![](animations/new_project.gif)
+
+When creating a DataWeave project, the DataWeave Extension plugin requires all dependencies to be loaded and indexed before it can search the libraries and make them available in your project. Visual Studio Code provides a notification bar at the bottom of the screen editor that shows when the dependencies are loading and when the indexing is complete.
+
+image::dataweave-extension-dependencies.png[Dependencies loading and indexing]
+//image::animations/loading.gif[Dependencies loading and indexing]
+// ![](animations/loading.gif)
+
+== Create a DataWeave Module
+
+DataWeave modules are `.dwl` files that define functions, variables, types, and namespaces. These files can't contain output directives, body expressions, or `---` separators between header and body sections. A DataWeave module enables you to share all the content defined in the `.dwl` file for reuse.
+
+To create a new DataWeave module, follow these steps:
+
+. In Visual Studio Code, launch *Command Palette* by pressing `Ctrl + Shift + P` (Windows, Linux) or `⇧⌘P` (macOS).
+. In the search bar, type `dataweave`.
+. Select *DataWeave: Create New Module*.
++
+
+image::dataweave-extension-createmodule.png[Command Palette with DataWeave: Create new module option selected]
+//image::animations/new_mapping.gif[Creating new Module using DataWeave Extension Plugin]
+// ![](animations/new_mapping.gif)
+
+== Test a DataWeave Module
+
+There are two ways to test your module: either by running a unit test or by creating a test mapping located on your `src/test/dw` folder as it should not be exposed to consumers of your module.
+
+The tests use functions of the modules `dw::test::Tests` and `dw::test::Asserts` provided by the artifact `org.mule.weave.wtf` of the
```

Comment:
Do we have descriptions of the functions in those modules? Will they be referenced in the testing framework doc maybe?

Reply:
The DW Testing framework doc does not have descriptions for  `dw::test::Tests` and `dw::test::Asserts`. Could you provide descriptions @martincousido? Thank you!

Reply:
Mmmm, I'm not sure if we need to provide the description in the doc. But the functions of those modules can be hover in the tests files .dwl and the user can see the description of what the different functions do. 
For example, `describedBy` "Defines a new test suite with the list of test cases."

Reply:
```suggestion
The tests use functions of the modules `dw::test::Tests` and `dw::test::Asserts` provided by the artifact `org.mule.weave:data-weave-testing-framework` of the
```
Renamed from wtf to data-weave-testing-framework. Not sure if `:` is used normally to separate groupId from artifactId.